### DISCUSSION TITLE
Struggling to get this running

### DISCUSSION THREAD

**Discussion by steinbitglis** [jasonwhite/rudolfs#60] [Q&A]
I have a docker setup with a running MinIO server.

The rudolfs docker container seems to be stuck on this:

```
, 2023-05-05T10:36:58.239Z INFO  rudolfs > Initializing storage...
, 2023-05-05T10:36:58.239Z INFO  rudolfs::storage::s3 > Connecting to S3 bucket 'git-lfs' at region 'us-west-1'
```

The server refuses connections at this point. I don't know where to begin to debug.

**Comment by steinbitglis** [comment]
The AWS_S3_ENDPOINT is `http://minio-1:9000` , which corresponds with my docker setup

**Comment by jasonwhite** [comment]
Sorry for the late reply. I just discovered that Gmail's spam filter was eating all of my GitHub notifications.

Are you using docker-compose? If not, then the Rudolfs container will not be able to connect to a port from the host or from another Docker container. Docker-compose will normally handle exposing the ports of other containers for you without manual configuration.

Also note that Rudolfs will retry the connection to S3 (or MinIO in this case) for quite a while until it is able to connect. It does this so that the MinIO server has time to start up. You should see it eventually timeout and an error message if it can't reach the MinIO server.
